gc/base/ParallelDispatcher.cpp

```diff
@@ -661,9 +660,18 @@ MM_ParallelDispatcher::contractThreadPool(MM_EnvironmentBase *env, uintptr_t new
 	if (newThreadCount < _threadCountMaximum) {
 		Trc_MM_ParallelDispatcher_contractThreadPool_Attempt();
 
+		_inShutdown = true;
+
 		omrthread_monitor_enter(_workerThreadMutex);
 
-		_inShutdown = true;
+		while (_workerThreadsReservedForGC) {
+			omrthread_monitor_wait(_workerThreadMutex);
+		}
+
+		/* Set the active parallel thread count to 1 */
+		/* This allows worker threads to cause a GC during their detach, */
+		/* making them the main in a single threaded GC. */
```

Comment:
Correct formatting for multi-line comments: https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#formatting-comments
```suggestion
		/* Set the active parallel thread count to 1. This allows 
		 * worker threads to cause a GC during their detach,
		 * making them the main in a single threaded GC.
		 */
```